services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py

```diff
@@ -51,19 +51,31 @@ async def get_user_project_access_rights(
             workspace_id=project_db.workspace_id,
             product_name=product_name,
         )
-        _user_project_access_rights = UserProjectAccessRights(
-            uid=user_id,
-            read=workspace.my_access_rights.read,
-            write=workspace.my_access_rights.write,
-            delete=workspace.my_access_rights.delete,
+        _user_project_access_rights_with_workspace = (
+            UserProjectAccessRightsWithWorkspace(
+                uid=user_id,
+                workspace_id=project_db.workspace_id,
+                read=workspace.my_access_rights.read,
+                write=workspace.my_access_rights.write,
+                delete=workspace.my_access_rights.delete,
+            )
         )
     else:
         _user_project_access_rights = (
             await db.get_pure_project_access_rights_without_workspace(
                 user_id, project_id
             )
         )
-    return _user_project_access_rights
+        _user_project_access_rights_with_workspace = (
+            UserProjectAccessRightsWithWorkspace(
+                uid=user_id,
+                workspace_id=None,
+                read=_user_project_access_rights.read,
+                write=_user_project_access_rights.write,
+                delete=_user_project_access_rights.delete,
+            )
+        )
+    return _user_project_access_rights_with_workspace
```

Comment:
ok.  In this context,  it makes sense indeed why you distinguish between `_user_project_access_rights` and `_user_project_access_rights_with_workspace`. But IMO it is a detail that should not be exposed outside of this function. Therefore I still think the model returned here should basically be a class suggesting "project access rights" without referring to "workspaces"

Reply:
Yes, currently you get access to the project based on different context (private workspace or shared workspace) it is useful to have this information in the output of this function as you can see I use it in some places.